src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -177,25 +177,49 @@ public class ChangeManagement implements Serializable {
         String credentialsId,
         String cmclientOpts = '') {
 
-        def args = [
-                '-tID', transportRequestId,
-                "\"$filePath\""
-            ]
+        // 1.) if needed in the context of fiori deploy (for 'ui5 build' this is needed):
+        //     create the config file file, eg. by calling the correponding wizzard
+        //     if possible we should locate that file somehere in a tmp folder in .pipeline
+        //     in order to avoid collisions with file from the project or in order to avoid
+        //     having that file in some build results (... zip).
+        //     revisit: how does this work when calling fiori deploy. There is no need for
+        //     providing a config file. But from the config file which is used for ui5 build
+        //     we fetch the credentials (not directy contained in that file, but exctracted via
+        //     that file from environmentVariables.
+
+        // 2.) create the call
+        def cmd = 'fiori deploy'
+
+        // 3.) execute the call in an appropirate docker container (fiori toolset) and evaluate the return code
+        //     or let the AbortException bubble up.
+        this.script.withCredentials([script.usernamePassword(
+            credentialsId: credentialsId,
+            passwordVariable: 'password',
+            usernameVariable: 'username')]) {
 
-        int rc = executeWithCredentials(
```

Comment:
We cannot reuse the executeWithCredentials method?

Reply:
Of course we can, let's see how much we have in common. We can either do it inside the `execWithCreds` method in a case  statement or here. Not sure it is easy to understand to prepare some parts, hand over to other method and prepare the rest. Basically `execWithCreds` was nice when we only had to support the cm_client. Started to get weired/questionable with supporting RFC. And having three entirely different kinds makes it IMO even more questionionable.

Reply:
I agree that we should strive for more readable code. It was tough for me to understand what is happening in this code...